packages/components/src/heading/test/index.tsx

```diff
@@ -51,8 +51,8 @@ describe( 'props', () => {
 				Code is Poetry
 			</Heading>
 		);
-		expect( container.firstChild.getAttribute( 'role' ) ).toBe( 'heading' );
-		expect( container.firstChild.getAttribute( 'aria-level' ) ).toBe( '3' );
+		expect( container.firstChild ).toHaveAttribute( 'role', 'heading' );
+		expect( container.firstChild ).toHaveAttribute( 'aria-level', '3' );
```

Comment:
Nit: If we are changing the previous uses of `container.firstChild` to `.children[ 0 ]` should we also do the same here?
```suggestion
		expect( container.children[ 0 ] ).toHaveAttribute( 'role', 'heading' );
		expect( container.children[ 0 ] ).toHaveAttribute( 'aria-level', '3' );
```

Reply:
If anything, I'd prefer we simplify in [this direction](https://github.com/WordPress/gutenberg/compare/refactor/heading-to-typescript...refactor/heading-to-typescript-tests?expand=1#diff-a4c73bed0ef8cc8f0e9883a24338969e5f48c16aab6b97fe297b9d887f19df0d), so we don't have to reach into containers directly. I think this style better matches the Testing Library paradigm. What do folks think?